### PR TITLE
Added extra tax bracket for II and CG

### PR DESCRIPTION
**Pull request by GoFroggyRun**
Per discussions in issue #132, this PR adds extra tax bracket on top of the current last bracket for individual income (II) and capital gain (CG), including AMT CG, respectively. 

The default value of these added parameters duplicates the existing top rates and thresholds, and thus there are no tax logic change. 

Test suites have been modified as well to incorporate these added parameters, and there're no change in any current diff files. I plan to add more tests regarding these extra parameters in another PR soon. 

@MattHJensen @martinholmer @feenberg Comments, concerns or remarks would be appreciated.

### CODE REVIEW

**Comment by GoFroggyRun** [comment]
@talumbau T.J., I'm confused by the Travis CI test failure, since I wasn't able to track the error message. Could you please advice what's going on?

**Comment by talumbau** [comment]
@GoFroggyRun there is something wrong with Travis CI this morning - at least, as far as I can tell. When I click on the link to look at the build logs, here is what I see:

screen shot 2016-08-08 at 9 41 37 am

File: taxcalc/functions.py [hunk 1]
```diff
@@ -540,9 +540,17 @@ def TaxGains(e00650, c01000, c04800, c23650, p23250, e01100, e58990,
         else:
             addtax = 0.
 
+        if c24540 > CG_thd_xtr[MARS - 1]:
+            addtax = (CG_rt_xtr - CG_rt3) * c24517 + addtax
+        elif c24540 <= CG_thd_xtr[MARS - 1] and _taxinc > CG_thd_xtr[MARS - 1]:
+            addtax = addtax + (CG_rt_xtr - CG_rt3) * \
+             min(_dwks21, _taxinc - CG_thd_xtr[MARS - 1])
```
Comment (talumbau) [inline taxcalc/functions.py]: I think this is the offending line. One way to make this PEP8 compliant would be to format this line in the same way that lines 538-539 are formatted.

Comment (GoFroggyRun) [inline taxcalc/functions.py]: Thanks for helping me locating the issue T.J. @talumbau!

**Comment by MattHJensen** [comment]
Thanks @GoFroggyRun! I wonder if Xtra is the right thing to append to the name, given that a user will need to modify both this rate and the rate below it in order to raise the top rate. Perhaps it should just be given the next number. E.g., `_AMT_CG_rt_4` instead of `_AMT_CG_rt_xtr`

**Comment by MattHJensen** [comment]
Something else I've been thinking about is whether we should add a function somewhere that automatically increases this rate if the rate below it is increased. 

That would maintain backward compatibility. Otherwise someone whose script raises the top rate will be broken by this PR. 

I'm thinking we should wait until @martinholmer is back from vacation before merging this PR, i trust you can still use it for your widget work in the meantime, @GoFroggyRun, right?

**Comment by feenberg** [comment]
On Tue, 9 Aug 2016, Matt Jensen wrote:

> 
> Something else I've been thinking about is whether we should add a function somewhere
> that automatically increases this rate if the rate below it is increased.
> 
> That would maintain backward compatibility. Otherwise someone whose script raises the
> top rate will be broken by this PR.

Doesn't this depend on the default threshold for the optional bracket? If 
the threshold is set at infinity, then rate 5 can be changed without 
changing rate 6.

**Comment by MattHJensen** [comment]
> If the threshold is set at infinity, then rate 5 can be changed without changing rate 6.

That is much more clever. Thanks @feenberg. cc @GoFroggyRun

**Comment by GoFroggyRun** [comment]
@MattHJensen said:
>
 I wonder if Xtra is the right thing to append to the name, given that a user will need to modify both this rate and the rate below it in order to raise the top rate. Perhaps it should just be given the next number.

Thanks for your comments. I have modified names of these extra parameters.

@feenberg said:
>
If the threshold is set at infinity, then rate 5 can be changed without changing rate 6.

Thank you so much for your suggestion Dan! This is a much more clever strategy. I have updated these thresholds, as well as reverted modifications on all tests.

**Comment by MattHJensen** [comment]
@GoFroggyRun, do you think the rates should be set to 1 or infinity, in parallel to the brackets being set to infinity, so that raising rt7 doesn't bump into rt8? 

@martinholmer, I think @GoFroggyRun and I have this covered now. I'll merge once we get these last changes. Thanks for the tip about the rates on the phone.

**Comment by GoFroggyRun** [comment]
@MattHJensen said:
>
do you think the rates should be set to 1 or infinity

Setting these two rates to one makes more sense to me.

**Comment by MattHJensen** [comment]
Thanks @GoFroggyRun. Merging.

**Pull request merged**